opm/input/eclipse/EclipseState/Grid/EclipseGrid.cpp

```diff
@@ -1959,10 +1962,175 @@ std::vector<double> EclipseGrid::createDVector(const std::array<int,3>& dims, st
+        }   
+    }
+    
+    std::size_t EclipseGrid::getTotalActiveLGR() const
+    {
+        std::size_t num_coarse_level =  getNumActive();
+        std::size_t num_fine_cells = 0;
+        std::size_t num_spanned_cells = 0;
+        for (const auto&  fine_cells : lgr_children_cells) {
+            num_fine_cells += fine_cells.getTotalActiveLGR();
+            num_spanned_cells += fine_cells.get_father_global().size();
+
+        }
+        return num_coarse_level + num_fine_cells - num_spanned_cells;
+    }  
+
+
+    void EclipseGrid::init_lgr_cells(const LgrCollection& lgr_input) {
+        save_all_lgr_labels(lgr_input);
+        create_lgr_cells_tree(lgr_input);
+        init_lgr_cells_index();
+        init_lgr_global_cells_index();
+    }
+    
+    void EclipseGrid::save_all_lgr_labels(const LgrCollection& lgr_input) {
+        all_lgr_labels.reserve(lgr_input.size()+1);
+        all_lgr_labels.push_back("GLOBAL");
+        for (std::size_t index = 0; index < lgr_input.size(); index++) {
+            all_lgr_labels.push_back(lgr_input.getLgr(index).NAME());
+        }    
+    }
+
+    void EclipseGrid::create_lgr_cells_tree(const LgrCollection& lgr_input) {                
+          auto IJK_global = [this](const auto& i_list, const auto& j_list, const auto& k_list){
+            if ((i_list.size() != j_list.size()) || (j_list.size() != k_list.size()) || (k_list.size() != i_list.size()) ){
+                 throw std::invalid_argument("Sizes are not compatible.");
+            }
+            std::vector<std::size_t> global_ind(i_list.size());
+            for (std::size_t index = 0; index < i_list.size(); index++) {              
+                global_ind[index] = this->getActiveIndex(i_list[index],j_list[index],k_list[index]);
+            }
+            return global_ind;
+        };
+        for (std::size_t index = 0; index < lgr_input.size(); index++) {
+            const auto lgr_cell = lgr_input.getLgr(index);
+            if (this->lgr_label.compare(lgr_cell.PARENT_NAME()) == 0)            {
+                lgr_grid = true;
+                auto [i_list, j_list, k_list] = lgr_cell.parent_cellsIJK();                
+                auto father_lgr_index = IJK_global(i_list, j_list, k_list);
+                lgr_children_cells.emplace_back(lgr_cell.NAME(), this->lgr_label, this->lgr_level, 
+                                                lgr_cell.NX(), lgr_cell.NY(), lgr_cell.NZ(), father_lgr_index);
+                
+                lgr_children_cells.back().create_lgr_cells_tree(lgr_input);              
+            }
+        }
+        std::sort(lgr_children_cells.begin(), lgr_children_cells.end(),[](const EclipseGridLGR& a, const EclipseGridLGR& b) {
+                return a.get_father_global()[0] < b.get_father_global()[0]; // Sort by another property
+        });
+        lgr_children_labels.reserve(lgr_children_cells.size());
+        for (auto lgr_cell : lgr_children_cells) {
```

Comment:
This copies the `lgr_children_cells` sequence.  Unless there's a reason to do so, and in that case please insert a comment for why we need to copy the sequence, we should normally write our loops as
```C++
for (const auto& lgr_cell : lgr_children_cells) {
    // ...
}
```
In this case, though, I'd probably use `std::transform()` instead&ndash;i.e.,
```C++
std::transform(lgr_children_cells.begin(), lgr_children_cells.end(),
               std::back_inserter(lgr_children_labels),
               [](const auto& lgr_cell) { return lgr_cell.lgr_label; });
```